### PR TITLE
Enable AITER ASM distributed FA testing in jax/torch

### PR DESCRIPTION
**Pull request by Micky774**
# Description

Please include a brief summary of the changes, relevant motivation and context.

Fixes # (issue)

## Type of change

- [ ] Documentation change (change only to the documentation, either a fix or a new content)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Infra/Build change
- [ ] Code refactoring

## Changes

Please list the changes introduced in this PR:

- Change A
- Change B

# Checklist:

- [ ] I have read and followed the [contributing guidelines](https://github.com/NVIDIA/TransformerEngine/blob/main/CONTRIBUTING.rst)
- [ ] The functionality is complete
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: ci/jax.sh
Comment (wangye805) [inline ci/jax.sh:73]: Need NVTE_CK_USES_FWD_V3=1 NVTE_CK_USES_BWD_V3=1 here

Comment (wangye805) [inline ci/jax.sh:77]: Also here

**Review by wangye805 — CHANGES_REQUESTED** [review]

**Comment by ipanfilo** [comment]
Which functionality not covered by existing tests does it cover?

**Review by wangye805 — APPROVED** [review]

**Comment by wangye805** [comment]
> Which functionality not covered by existing tests does it cover?

Previously our jax and pytorch distributed fused-attn only enables v2 ck backends, not v3

**Comment by ipanfilo** [comment]
> > Which functionality not covered by existing tests does it cover?
> 
> Previously our jax and pytorch distributed fused-attn only enables v2 ck backends, not v3

Yes, but does it run different fused attn backend configs/kernels then non distributed ones? Or there is functionality concern of coexistence of them with RCCL?

**Comment by wangye805** [comment]
> > > Which functionality not covered by existing tests does it cover?
> > 
> > 
> > Previously our jax and pytorch distributed fused-attn only enables v2 ck backends, not v3
> 
> Yes, but does it run different fused attn backend configs/kernels then non distributed ones? Or there is functionality concern of coexistence of them with RCCL?

In the distributed fused-attn (CP) pytest suite, the reference run is usually a single-GPU fused-attn with full seqlen (for example sq=skv=8192) using the default attn backend. The target run decomposes the single full-size fused-attn into 4 or 8 smaller fused-attn (for example, sq=sk=4096), runs those smaller fused-attn instances using the default backend and then "glue" the results in the CP way. 

In my option, why we need to enable v3 for distributed fused-attn:
1). The decomposition may create new fused-attn configs not covered by our single GPU fused-attn pytests
2). The "glue" process actually tested the softmax_lse generated in fwd pass. If the softmax lse results are not correct, the glued results will be wrong. And our single GPU fused-attn pytest do not test softmax_lse at all.

File: ci/jax.sh [hunk 1]
```diff
@@ -71,13 +71,16 @@ run_test_config_mgpu() {
     *0.4.35*)
         # Workaround for distributed tests hang with xla_flag
  	    XLA_FLAGS="--xla_gpu_enable_nccl_comm_splitting=false" run 3 test_distributed_fused_attn.py -k 'not test_context_parallel_ring_attn'
+ 	    XLA_FLAGS="--xla_gpu_enable_nccl_comm_splitting=false" NVTE_CK_USES_FWD_V3=1 NVTE_CK_USES_BWD_V3=1 run 3 test_distributed_fused_attn.py -k 'not test_context_parallel_ring_attn'
```
Comment (ipanfilo) [inline ci/jax.sh]: This will run it with AOTriton too

**Review by ipanfilo — CHANGES_REQUESTED** [review]

File: ci/jax.sh (hunk 1 above)
Comment (Micky774) [reply ci/jax.sh]: Updated with a guard in the JAX ci script

**Comment by wenchenvincent** [comment]
@Micky774 Could you rebase upon latest dev to incorporate the hot fix for the core sgpu tests?

File: ci/jax.sh (hunk 1 above)
Comment (ipanfilo) [reply ci/jax.sh]: With those changes env variables are not seen by run method - they are applied to test call only.
Using run_default_fa_lbl. All V3 calls should be labelled with "v3" to distinct them from regular test_distributed_fused_attn call

**Comment by wenchenvincent** [comment]
@ipanfilo Could you check if all your comments have been addressed?

**Review by wenchenvincent — APPROVED** [review]

**Review by ipanfilo — CHANGES_REQUESTED** [review]

**Comment by ipanfilo** [comment]
This PR interferes with https://github.com/ROCm/TransformerEngine/pull/372 which is merged already

**Comment by Micky774** [comment]
> This PR interferes with #372 which is merged already

This PR can be outright closed right, since we're now testing w/ V3 enabled by default?

**Comment by ipanfilo** [comment]
> This PR can be outright closed right, since we're now testing w/ V3 enabled by default?
Yes, it seems so.
